app/src/ui/changes/changes-list.tsx

```diff
@@ -432,6 +432,7 @@ export class ChangesList extends React.Component<
               : 'Summary (required)'
           }
           singleFileCommit={singleFileCommit}
+          key={this.props.repository.id}
```

Comment:
What's the context for adding this here? Did you uncover something that we weren't doing well before?

Reply:
> uses a key to manage un-mounting and mounting a new CommitMessage

the key here forces the component to be remounted when the key changes, and only updates when the key is the same. this simplifies the components internal logic around unmounting and saving the draft commit message to the stores